functions/heli/fn_initHeli.sqf

```diff
@@ -0,0 +1,142 @@
+			_args params ["_hangar", "_newType"];
+
+			// Get the nearest chopper
+			_heli = [_hangar] call GSRI_fnc_getHeli;
+
+			// Select formattable string depending on wether there is already a chopper or not
+			_displayName = ["Remplacer %1 par %2", "Demander %2"] select (isNull _heli);
+
+			// Proper formatting of displayName and action parameters
+			_actionData set [1, format [_displayName, [_heli] call GSRI_fnc_minifyName, [_newType] call GSRI_fnc_minifyName]];
+		};
+		_condition = {
+			params["_t", "_p", "_args"];
+
+			// Retrieving action parameters
+			_args params ["_hangar", "_newType"];
+
+			_heli = [_hangar] call GSRI_fnc_getHeli;
+
+			// If names are the same, do not display
+			(([_newType] call GSRI_fnc_minifyName) != ([_heli] call GSRI_fnc_minifyName))
+		};
+		_action = [format["action%1", _display],_display,"",GSRI_fnc_spawnHeli,_condition,{},[_hangar, _x],"",2,[false, false, false, false, false], _modifier] call ace_interact_menu_fnc_createAction;
+		[_handle, 0, ["ACE_MainActions"], _action] call ace_interact_menu_fnc_addActionToObject;
+	};
+} forEach _types;
+
+// Hangar cleaning action
+_modifier = {
+    params ["_t", "_p", "_args", "_actionData"];
+
+	// Retrieving action parameters
+	_args params ["_hangar"];
+
+	// Get the nearest chopper
+	_heli = [_hangar] call GSRI_fnc_getHeli;
+
+	// Proper formatting of displayName and action parameters
+    _actionData set [1, format ["Supprimer %1", [_heli] call GSRI_fnc_minifyName]];
+};
+_condition = {
+	params ["_t", "_p", "_args"];
+	_args params ["_hangar"];
+	!isNull ([_hangar] call GSRI_fnc_getHeli)
+};
+_statement = {
+	params ["_t", "_p", "_args"];
+	_args params ["_hangar"];
+	_heli = ([_hangar] call GSRI_fnc_getHeli);
+	["HeliDelete", [getText (configFile >> "CfgVehicles" >> typeOf _heli >> "displayName")]] call BIS_fnc_showNotification;
+	deleteVehicle _heli;
+};
+_action = ["actionClear","Supprimer","",_statement,_condition,{},[_hangar],"",2,[false, false, false, false, false], _modifier] call ace_interact_menu_fnc_createAction;
+[_handle, 0, ["ACE_MainActions"], _action] call ace_interact_menu_fnc_addActionToObject;
+
+// FRIES mounting action
+_condition = {
+	params ["_t", "_p", "_args"];
+	_args params ["_hangar"];
+	_heli = [_hangar] call GSRI_fnc_getHeli;
+	(isNumber (configFile >> "CfgVehicles" >> typeOf _heli >> "ace_fastroping_enabled") && isNull (_heli getVariable ["ace_fastroping_FRIES", objNull]));
+};
+_statement = {
+	params ["_t", "_p", "_args"];
+	_args params ["_hangar"];
+	_heli = [_hangar] call GSRI_fnc_getHeli;
+	["HeliFRIES", [getText (configFile >> "CfgVehicles" >> typeOf _heli >> "displayName")]] call BIS_fnc_showNotification;
+	[_heli] remoteExecCall ["ace_fastroping_fnc_equipFRIES", 2];
+};
+_action = ["actionFRIES","Equiper FRIES","",_statement,_condition,{},[_hangar]] call ace_interact_menu_fnc_createAction;
+[_handle, 0, ["ACE_MainActions"], _action] call ace_interact_menu_fnc_addActionToObject;
+
+// Place helicopter on rear deck
```

Comment:
Comment denotes a good candidate for an external function

Reply:
this is a temporary function and will not be externalized